linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/domain/DataType.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.storage.domain;
+
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class DataType {
+  private static Logger logger = LoggerFactory.getLogger(DataType.class);
+
+  public static final String NULL_VALUE = "NULL";
+  public static final String LOWCASE_NULL_VALUE = "null";
+
+  // TODO Change to fine-grained regular expressions(改为精细化正则表达式)
+  public static final Pattern DECIMAL_REGEX = Pattern.compile("^decimal\\(\\d*\\,\\d*\\)");
+
+  public static final Pattern SHORT_REGEX = Pattern.compile("^short.*");
+  public static final Pattern INT_REGEX = Pattern.compile("^int.*");
+  public static final Pattern LONG_REGEX = Pattern.compile("^long.*");
+  public static final Pattern BIGINT_REGEX = Pattern.compile("^bigint.*");
+  public static final Pattern FLOAT_REGEX = Pattern.compile("^float.*");
+  public static final Pattern DOUBLE_REGEX = Pattern.compile("^double.*");
+
+  public static final Pattern VARCHAR_REGEX = Pattern.compile("^varchar.*");
+  public static final Pattern CHAR_REGEX = Pattern.compile("^char.*");
+
+  public static final Pattern ARRAY_REGEX = Pattern.compile("array.*");
+
+  public static final Pattern MAP_REGEX = Pattern.compile("map.*");
+
+  public static final Pattern LIST_REGEX = Pattern.compile("list.*");
+
+  public static final Pattern STRUCT_REGEX = Pattern.compile("struct.*");
+
+  public String typeName;
+  private int javaSQLType;
+
+  public DataType(String typeName, int javaSQLType) {
+    this.typeName = typeName;
+    this.javaSQLType = javaSQLType;
+  }
+
+  @Override
+  public String toString() {
+    return typeName;
+  }
+
+  public String getTypeName() {
+    return typeName;
+  }
+
+  public int getJavaSQLType() {
+    return javaSQLType;
+  }
+
+  public static DataType toDataType(String dataType) {
+    if (dataType.equals("void") || dataType.equals("null")) {
+      return new NullType();
```

Comment:
Is it possible to adjust to obtain the string first through the regularization, and then judge the string through the switch case

Reply:
Thank you, I still need to continue the review, you can not modify it now